example/echo.re

```diff
@@ -1,27 +1,34 @@
 let handler: AwsLambda.APIGatewayProxy.handler =
   (event, _context, cb) => {
+    open AwsLambda.APIGatewayProxy;
     let parameter =
-      Js.Null.toOption(event##queryStringParameters)
+      event
+      |. Event.queryStringParameters
       |> Js.Option.andThen((. params) => Js.Dict.get(params, "userid"));
-    switch parameter {
+    switch (parameter) {
     | Some(userid) => Js.log2("executing lambda for", userid)
     | None => Js.log("executing lambda for anonymous user")
     };
     let result =
-      switch (Js.Null.toOption(event##body)) {
-      | None =>
+      switch (event |. Event.body, event |. Event.isBase64Encoded) {
+      | (None, false) =>
         Js.log("error: no body available in the request");
-        AwsLambda.APIGatewayProxy.result(
+        result(
           ~body=`Plain({|{"status": "no body available in the request"}|}),
           ~statusCode=400,
-          ()
+          (),
         );
-      | Some(body) => {
-          "body": body,
-          "statusCode": 200,
-          "headers": Js.Nullable.null,
-          "isBase64Encoded": Js.Nullable.return(event##isBase64Encoded)
-        }
+      | (None, true) =>
+        Js.log("error: no body available in the request");
+        result(
+          ~body=`Base64({|{"status": "no body available in the request"}|}),
```

Comment:
this part doesn't look correct. `{|{"status": "no body available in the request"}|}` is plain content, not base64 encoded. So there is no need to switch on `event |. Event.isBase64Encoded` and duplicate the branches.

Reply:
The old example wasn't passing the type check so I fixed it up. You're right though this content is not `base64` so I'll convert that in the example